app/controllers/users_controller.rb

```diff
@@ -4,34 +4,54 @@ class UsersController < ApplicationController
   before_action :require_sign_in
   load_and_authorize_resource
   load_and_authorize_resource :vita_partner, collection: [:edit, :update], parent: false
-
+  before_action :authorize_vita_partner_updates, only: :update
   layout "admin"
 
-  def profile
-  end
+  def profile;end
 
   def index
     @users = @users.includes(:vita_partner)
   end
 
   def edit
+    @user.vita_partner_id = @user.memberships.find_by(role: "member")&.vita_partner_id
+    @user.supported_organization_ids = @user.memberships.where(role: "lead").pluck(:vita_partner_id)
   end
 
   def update
-    authorize!(:manage, @vita_partners.find(user_params[:vita_partner_id]))
-    return render :edit unless @user.update(user_params)
-
+    authorize_vita_partner_updates
+    return render :edit unless update_user
     redirect_to edit_user_path(id: @user), notice: I18n.t("general.changes_saved")
   end
 
   private
+  # To leave the view as-is but allow the ability to save as memberships, we're doing the simplest thing possible:
+  # blowing away and recreating memberships on every update.
+  def update_user
+    ApplicationRecord.transaction do
+      @user.memberships.destroy_all && @user.update!(user_params.merge(memberships_attributes: memberships))
+    end
+
+  rescue ActiveRecord::RecordInvalid
+    logger.info("User update unexpectedly failed. Rolling back transaction.")
+  end
+
+  def memberships
+    supported_org_ids = (params[:user][:supported_organization_ids] || []).reject(&:empty?)
+    memberships = supported_org_ids.map { |so| { user_id: @user.id, vita_partner_id: so, role: "lead"} }
+    memberships.push({user_id: @user.id, vita_partner_id: params[:user][:vita_partner_id], role: "member"}) if params[:user][:vita_partner_id].present?
+    memberships
+  end
+
+  def authorize_vita_partner_updates
+    changing_vita_partner_ids = [params[:user][:vita_partner_id], params[:user][:supported_organization_ids]].flatten.select(&:present?)
+    changing_vita_partner_ids.each { |id| authorize!(:manage, @vita_partners.find(id)) }
+  end
```

Comment:
If we are having to write custom logic for handling params and the object creation process for multiple models, I'd suggest using a `Form` abstraction. Perhaps a `UserUpdateForm` that you can add a corresponding unit test file for. That would allow us to separate out the permissions and authorization responsibilities of the controller from the param restructuring and model updating that the form can handle itself. `TakeActionForm` also handles a bunch of different models, so it might be a good comparison to start.

Reply:
This is the shim that we plan on removing and replacing with a simplified structure once we decide on what the updated UI should look like, so a refactor into a form object for "bridge" code doesn't seem like the write abstraction.